src/notifications/Manager.cpp

```diff
@@ -0,0 +1,90 @@
+#include "notifications/Manager.h"
+
+#include "Cache.h"
+#include "EventAccessors.h"
+#include "Logging.h"
+#include "MatrixClient.h"
+#include "Utils.h"
+
+#include <QFile>
+#include <QImage>
+#include <QStandardPaths>
+
+#include <mtxclient/crypto/client.hpp>
+
+QString
+NotificationsManager::cacheImage(const mtx::events::collections::TimelineEvents &event)
```

Comment:
Could you make this a generic function, that is shared between all the image downloads we do? I don't feel good duplicating this again. Maybe give it a callback for when the download completed or so.